src/mumble/MainWindow.cpp

```diff
@@ -3218,6 +3218,10 @@ void MainWindow::serverConnected() {
 		Global::get().sh->setSelfMuteDeafState(Global::get().s.bMute, Global::get().s.bDeaf);
 	}
 
+	if (Global::get().s.bTCPCompat) {
+		Global::get().sh->forceTcpOnlyMode(true);
+	}
```

Comment:
Wouldn't it make more sense to let the ServerHandler check for this setting itself when it is about to establish a connection? That way we would exclude any potential weird issues arising due to bad timings where the connection has already been established but this slot has not yet executed and thus the handler still thinks it is supposed to use UDP...

Reply:
Yes, since it is a crucial parameter which is best be configured once during connection establishment, as you said.

I think the best place to put it would be in the slot `void ServerHandler::serverConnectionConnected();`. Please let me know your opinion on that?

Reply:
Yes, I would say that would be a good fit :+1: